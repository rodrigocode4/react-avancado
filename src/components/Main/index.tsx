import * as S from './styled'

const Main = () => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Imagem de um átomo dentro de um hexágono."
    />
    <S.Title>React Avançado</S.Title>
    <S.Description>
      Typescript, ReactJS, NextJS e Styled Components
    </S.Description>
    <S.Ilustration
      src="/img/hero-illustration.svg"
      alt="Um desenvolvedor de frente para uma tela com código."
    />
  </S.Wrapper>
)

export default Main
