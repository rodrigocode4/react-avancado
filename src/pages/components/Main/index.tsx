const Main = () => (
  <main>
    <h1>Rodrigo Oliveira JS</h1>
  </main>
)

export default Main
