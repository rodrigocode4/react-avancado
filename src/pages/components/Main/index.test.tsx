import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main/', () => {
  it('Deve renderizar o heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /rodrigo oliveira js/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
