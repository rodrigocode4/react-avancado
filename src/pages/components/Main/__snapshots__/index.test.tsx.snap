// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/ Deve renderizar o heading 1`] = `
<main>
  <h1>
    Rodrigo Oliveira JS
  </h1>
</main>
`
